google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -738,11 +739,16 @@ private static WriteLogEntriesRequest writeLogEntriesRequest(
       LoggingOptions serviceOptions,
       Iterable<LogEntry> logEntries,
       Map<Option.OptionType, ?> options) {
-    String projectId = serviceOptions.getProjectId();
     WriteLogEntriesRequest.Builder builder = WriteLogEntriesRequest.newBuilder();
-    String logName = LOG_NAME.get(options);
+    String projectId = null;
+
+    LogName logName =
+        getLogName(
+            serviceOptions.getProjectId(), LOG_NAME.get(options), LOG_DESTINATION.get(options));
+
```

Comment:
Would you define criteria when a block of code in one method should be branched to a separate method beside the case when this code is reused/called in/from more than one place? It would be great to have a shared guideline for this so we can have a homogeneously looking code.

Reply:
I believe 5-15 lines including parameters is more than enough (so this way entire function can be visible on small laptop screen). Some more good answers you can find [here](https://softwareengineering.stackexchange.com/questions/133404/what-is-the-ideal-length-of-a-method-for-you).